### PR TITLE
Michael/use pouch

### PR DESCRIPTION
**Pull request by pzhine**

### CODE REVIEW

File: packages/databyss-services/session/clientStorage.js [hunk 1]
```diff
@@ -26,3 +26,15 @@ export function getAccountId() {
 export function deleteAccountId() {
   localStorage.removeItem('account')
 }
+
+export function setDefaultPageId(value) {
```
Comment (pzhine) [inline packages/databyss-services/session/clientStorage.js:30]: Why are we storing this locally now?

File: packages/databyss-services/database/_helpers.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import { PatchBatch } from '../interfaces/Patch'
+import { runPatches } from './patches'
+import { DocumentType, DbPage, MangoResponse } from './interfaces'
+import { PageHeader } from '../interfaces/Page'
+import { initSelection, initBlock, initPage } from './initialState'
+
+export const populatePage = async (_id: string): Promise<Page | null> => {
+  const _response = await db.find({
+    selector: {
+      documentType: 'PAGE',
```
[the remark this answers is not in the document]
Comment (msenejoa) [reply packages/databyss-services/database/_helpers.ts]: `_type` is not allowed on PouchDB as a property

File: packages/databyss-data/schemas/blockRelationSchema.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    },
+    modifiedAt: {
+      type: 'number',
+    },
+    createdAt: {
+      type: 'number',
+    },
+  },
+  required: [
+    '_id',
```
Comment (pzhine) [inline packages/databyss-data/schemas/blockRelationSchema.ts:59]: Similarly, we should have a mixin for the `required` array so we can do
```
required: [
  ...pouchDocRequired,
  block,
  ...
]
```

File: packages/databyss-data/schemas/entrySchema.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          items: {
+            type: 'string',
+          },
+        },
+      },
+    },
+    _id: {
+      type: 'string',
+    },
+    type: {
```
Comment (pzhine) [inline packages/databyss-data/schemas/entrySchema.ts:27]: Following the mixin pattern, we should have a `blockSchema` mixin for schemas that share the `block` base class.
```
export const blockSchema = {
  type: {
    type: 'string',
  },
  text: {
    $ref: 'text',
  },
  account: {
    type: 'string',
  },
}
```

File: packages/databyss-data/schemas/test.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (pzhine) [inline packages/databyss-data/schemas/test.json:1]: Can this be cleaned up or moved?

File: packages/databyss-editor/lib/clipboardUtils/deleteBlocksAtSelection.ts [hunk 6]
```diff
@@ -124,6 +124,7 @@ export default (draft: EditorState) => {
   // collapse selection
   const _cursor = { offset: anchor.offset, index: anchor.index }
   draft.selection = {
+    $type: DocumentType.Selection,
```
Comment (pzhine) [inline packages/databyss-editor/lib/clipboardUtils/deleteBlocksAtSelection.ts:127]: If we have to use `$type` or `DocumentType` outside of `@databyss-org/data` we've done something wrong. It is meant as a field internal to our database library for managing types. Let's discuss why we are using this here and some alternatives.

File: packages/databyss-services/interfaces/Block.ts [hunk 7]
```diff
@@ -2,6 +2,13 @@ import { BlockRelation } from '@databyss-org/editor/interfaces/index'
 import { SelectOption } from './UI'
 import { Text } from './Text'
 
+export enum DocumentType {
```
Comment (pzhine) [inline packages/databyss-services/interfaces/Block.ts:5]: See note above about `DocumentType` scope. This should be in `@databyss-org/data`

File: packages/databyss-ui/components/Notify/NotifyProvider.js [hunk 8]
```diff
@@ -47,7 +43,7 @@ class NotifyProvider extends React.Component {
   constructor(props) {
     super(props)
     startBugsnag(props.options)
-    this.shouldCheckOnlineStatus = props.shouldCheckOnlineStatus
+    // this.shouldCheckOnlineStatus = props.shouldCheckOnlineStatus
```
Comment (pzhine) [inline packages/databyss-ui/components/Notify/NotifyProvider.js:46]: TODO: restore online status checking and inspect pouchdb sync state so we can inform the user of sync progress

File: packages/databyss-ui/components/PageContent/PageHeader.js [hunk 9]
```diff
@@ -34,13 +34,14 @@ const PageHeader = forwardRef(({ pageId, onNavigateDownFromHeader }, ref) => {
...
   const throttledAutosave = useCallback(
-    throttle((val) => {
+    debounce((val) => {
       const _pageData = {
         name: val.textValue ? val.textValue : noPageTitle,
         _id: pageId,
       }
       setPageHeader(_pageData)
-    }, process.env.SAVE_PAGE_THROTTLE),
+      // TODO: CHANGE THIS BACK TO process.env.SAVE_PAGE_THROTTLE
```
Comment (pzhine) [inline packages/databyss-ui/components/PageContent/PageHeader.js:43]: Can we do this now?

**Pull request merged**